app/database/operator/server_data_operator/handlers/post.test.ts

```diff
@@ -309,6 +312,58 @@ describe('*** Operator: Post Handlers tests ***', () => {
         expect(spyOnHandlePostsInChannel).toHaveBeenCalledTimes(1);
         expect(spyOnHandlePostsInChannel).toHaveBeenCalledWith(linkedPosts.slice(0, 3), actionType, true);
     });
+
+    it('=> HandlePosts: should properly parse metadata when the metadata is a string', async () => {
+        // @ts-expect-error notifications are sending metadata as a string, so we need to handle that case
+        const updatedPosts: Post[] = posts.map((post) => {
+            if (post.metadata && typeof post.metadata === 'object') {
+                return {
+                    ...post,
+                    metadata: JSON.stringify(post.metadata),
+                };
+            }
+            return post;
+        });
```

Comment:
This looks off for a test. We know the input data (and rely on the input data to be of certain way on the expect), so we can be a bit more direct here:
```
const updatedPosts = [{
    ...posts[0],
    metadata: JSON.stringify(posts[0].metadata),
}];
```
0/5 though.

Reply:
Agreed, I wanted the entire posts.  Only the first has metadata.

Actually, I should be writing test to test reaction as well.  Since this is related.

Reply:
I have directly created a new Posts array with only 1 post, in which the metadata is a string.

I have assert that all three exists, by spying on the 3 methods that "handle" files, emojis and reactions.

That should give enough confident that a string with can be parsed into JSON object.  However, I'm not creating a test if the metadata contains a string with invalid JSON text.  It's currently

```
    } catch {
        // Do nothing
    }
```

So, open to suggestion on what to do if the metadata is invalid parseable json string.